staging/src/k8s.io/apiserver/pkg/cel/environment/base.go

```diff
@@ -43,7 +44,14 @@ import (
 // desirable because it means that CEL expressions are portable across a wider range
 // of Kubernetes versions.
 func DefaultCompatibilityVersion() *version.Version {
-	return version.MajorMinor(1, 29)
+	ver := utilversion.DefaultEffectiveVersionRegistry.EffectiveVersionForOrRegister(
+		utilversion.ComponentGenericAPIServer, utilversion.K8sDefaultEffectiveVersion(),
```

Comment:
Ok, this instance of `K8sDefaultEffectiveVersion` appears to use component-base, which recommends using

```go
func Get() apimachineryversion.Info {
	// These variables typically come from -ldflags settings and in
	// their absence fallback to the settings in ./base.go
	return apimachineryversion.Info{
```

if I use their recommendation and set the version of my aggregated apiserver to 4.18, will the CEL environment be configured as I expect for 1.31?

Reply:
If aggregated apiserver has registered an effective version of 4.18, then the CEL environment be configured for 4.17. If you want it to be 1.31, then aggregated apiserver's effective version should register under a different component name.

Reply:
good point. Changed it to hardcoded KubeVersion.